src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/custom.py

```diff
@@ -97,3 +98,90 @@ def _validate_start_time(start_time, end_time):
+                           timezone: 1970-01-01T00:00:00Z, 1970-01-01T00:00:00-0500
+    :param str end_time: The end time of the query. In ISO format with explicit indication of
+                         timezone: 1970-01-01T00:00:00Z, 1970-01-01T00:00:00-0500
+    :param str caller: The caller to look for when querying
+    :param str status: The status value to query (ex: Failed)
+    :param str max_events: The maximum number of records to be returned by the command
+    :param str select: The list of event names
+    '''
+    present_switch = 0
+    for param in [correlation_id, resource_group, resource_uri, resource_provider]:
+        present_switch = present_switch + (1 if param else 0)
+
+    if present_switch > 1:
+        raise CLIError("'--correlation-id', '--resource-group', '--resource-uri'"
+                       " and '--resource-provider' are optional and mutually exclusive.")
+
+    formatter = "eventTimestamp ge {} and eventTimestamp le {}"
+    odata_filters = _validate_time_range_and_add_defaults(start_time, end_time,
+                                                          formatter=formatter)
+
+    if correlation_id:
+        odata_filters = _process_prameter(odata_filters, 'correlation_id',
+                                          correlation_id, 'correlationId')
+    elif resource_group:
+        odata_filters = _process_prameter(odata_filters, 'resource_group',
+                                          resource_group, 'resourceGroupName')
+    elif resource_uri:
+        odata_filters = _process_prameter(odata_filters, 'resource_uri',
+                                          resource_uri, 'resourceId')
+    elif resource_provider:
+        odata_filters = _process_prameter(odata_filters, 'resource_provider',
+                                          resource_provider, 'resourceProvider')
+    if caller:
+        odata_filters = _process_prameter(odata_filters, 'caller',
+                                          caller, 'caller')
+    if status:
+        odata_filters = _process_prameter(odata_filters, 'status',
+                                          status, 'status')
+    if max_events:
+        max_events = int(max_events)
+
+    select_filter = _activity_logs_select_filter_builder(select)
+    activity_logs = client.list(filter=odata_filters, select=select_filter)
+    return _limit_results(activity_logs, max_events)
+
+
+def _activity_logs_select_filter_builder(events=None):
+    '''Build up select filter string from events
+    '''
+    if events:
+        return ' , '.join(events)
+    return None
+
+
+def _process_prameter(default_filter, field_name, field_value, field_label):
+    if not field_value:
+        raise CLIError('Value for {} can not be empty.'.format(field_name))
+
+    return _add_condition_if_present(default_filter, field_label, field_value)
+
+
+def _add_condition_if_present(default_filter, field_label, field_value):
+    if not field_value:
+        return default_filter
```

Comment:
Yup i can just say `_add_condition` and format as long as we have `field_value` because function `_process_prarameter` would have thrown if `field_value` is not there.